File: academic-system-java/src/main/java/org/example/academic/system/service/ClassService.java
package org.example.academic.system.service;

import org.example.academic.system.exception.AcademicSystemException;
import org.example.academic.system.model.AcademicClass;
import org.example.academic.system.model.AcademicSystem;

public class ClassService {
    private final AcademicSystem academicSystem;

    // Injeta o modelo central do sistema para acesso às turmas
    public ClassService(AcademicSystem academicSystem) {
        this.academicSystem = academicSystem;
    }

    // Cria e registra uma nova turma, garantindo que o código ainda não exista no sistema
    public void registerClass(String code, String title) {
        AcademicClass academicClass = new AcademicClass(code, title);

        if (academicSystem.findClassByCode(academicClass.getCode()).isPresent()) {
            throw new AcademicSystemException("Ja existe uma turma com este codigo.");
        }

        academicSystem.addClass(academicClass);
    }
}

File: academic-system-python/academic_system/service/class_service.py
from ..exception.academic_system_exception import AcademicSystemException
from ..model.academic_class import AcademicClass
from ..model.academic_system import AcademicSystem


class ClassService:
    def __init__(self, academic_system: AcademicSystem):
        self._academic_system = academic_system

    # Cria e registra uma nova turma, garantindo que o código ainda não exista no sistema
    def register_class(self, code: str, title: str) -> None:
        if self._academic_system.find_by_code(code):
            raise AcademicSystemException(f"Turma com codigo '{code}' ja existe.")
        academic_class = AcademicClass(code, title)
        self._academic_system.add_class(academic_class)
